src/common/txsplain.js

```diff
@@ -708,6 +714,27 @@ angular.module('txsplain', [])
           return html
         }
 
+        // renderEnableAmendment
+        function renderEnableAmendment() {
+          if (tx.tx.Flags & 0x00010000) {
+            //
+            if (tx.meta.AffectedNodes[0].ModifiedNode.FinalFields.Majorities[0].Majority.Amendment === tx.tx.Amendment) {
+              // close time is where we expect it in the metadata. use that.
+              var close_time = tx.meta.AffectedNodes[0].ModifiedNode.FinalFields.Majorities[0].Majority.CloseTime
+              var expected_date = moment.unix(close_time + RIPPLE_EPOCH).utc().add(14, 'days')
+            } else {
```

Comment:
I'm curious why there are 2 separate possibilities here, where you are getting the date in different ways.

Reply:
The first way _should_ always work to get the time the transaction gained a majority. But it's pretty specific (digging deep into the metadata tree) so I added a fallback of using the current ledger close time just in case that doesn't work for some reason. Basically I'm just nervous about always assuming that reading `tx.meta.AffectedNodes[0].ModifiedNode.FinalFields.Majorities[0].Majority.CloseTime` will always work.

Reply:
its probably more likely to error than get to the `else`, as if any of the nested fields are missing/different than expected, it will error with something like `cannot read 'Majorities' of undefined` - but to ensure there is no chance of error, you'd have to do `if(tx && tx.meta & tx.meta.AffectedNodes && tx.meta.AffectedNodes[0] ........` or use try/catch